package tutorial;


message Operation {
	enum OperationType {
		REGISTER = 0;
		QUERY = 1;
		DISTRIBUTE = 2;
		RETURN = 3;
		FINISH = 4;
	}

	required OperationType operation = 1;

	optional uint32 port = 2 [default = 0];
	
	optional bool query = 3 [default = false];

	message DistributeTask {
		required string operation_num_type = 1;
		required string operation_num_one = 2;
		required string operation_num_two = 3;
		required string operation_label = 4;
	}
	repeated DistributeTask task = 4;

	message Result {
		required string result_type = 1;
		required string result_value = 2;
	}
	repeated Result result = 5;

	optional bool task_finished = 6 [default = false];

}

